tools/checklocks/state.go

```diff
@@ -182,9 +182,51 @@ func (l *lockState) count() int {
 	return len(l.lockedMutexes)
 }
 
-// isCompatible returns true if the states are compatible.
+// isCompatible returns true if the states hold the same locks.
+// The nil state is the lock superset and is compatible with any other
+// lock state.
 func (l *lockState) isCompatible(other *lockState) bool {
-	return l.isSubset(other) && other.isSubset(l)
+	if l == nil || other == nil {
+		return true
+	} else {
```

Comment:
nit: you can probably do away with the else branch since then branch has a return. However, I would just be consistent with the rest of the code in the repo.